### ISSUE TITLE
Differencing a cubic sphere sweep - artifact problem!

### ISSUE THREAD

**Issue by wfpokorny** [POV-Ray/povray#287]
(also sphere_sweep related: #59, #147 #287)

As reported by Yadgar via the povray.binaries.images newsgroup:

http://news.povray.org/povray.binaries.images/thread/%3C5905c94a%241%40news.povray.org%3E/

Artifacts exist for both b_spline and cubic sphere_sweeps in 3.7.0 through the current 3.7.2 master. The result looks to be correct when run against POV-Ray 3.6.1 so related to some change over time. 

See zip:
[Yadgar_spheresweep_differenceArtifact_043017.zip](https://github.com/POV-Ray/povray/files/966795/Yadgar_spheresweep_differenceArtifact_043017.zip)

**Comment by LeForgeron** [comment]
I tested with hg-povray and it is fine there (at least the tmp372.pov file)

Actually, it is also ok on my master head, but both stable and official master have problem.

Which leads me to conclude: read the note about the patch of my master, for sphere_sweep:
* sphere sweep: less noise under some angles
* sphere sweep: UV mapping
* sphere sweep: removed the preliminary workaround for FS#81 (not effective, was adding noise)
* sphere sweep: added the suggested fix for FS#243 (negative uniform scaling of untransformed object should have effect on radius as absolute factor)


From what I would keep in the sphsweep.cpp, previously in source/backend/shape (now spheresweep.cpp in source/core/shape ):
const DBL ZERO_TOLERANCE        = 1.0e-1;// less noise when near 1 than near 0 [JG]

instead of 

const DBL ZERO_TOLERANCE        = 1.0e-4;

**Comment by wfpokorny** [comment]
Thanks for the information Jérôme. I spent some of yesterday collecting past problematic test scenes so I can test changes such as those you mention. 

After the port of FS#243 to github as #285 I created a pull request #286 with Simon's fix against the 3.7.1 release branch. I agree with you & Simon that one a safe solid fix.

In what I have run thus far I agree too the 3.7 patch made for FS#81 is causing more problems than it is solving. If that holds, I think we should remove this work around for the 3.7.1 release. 

I'd seen your "less noise under certain angles change" - I think in the FS#81 comment stream - but, it looked in your comments  like you'd not run some of the known problem cases. I'll try it against my sphere_sweep collection, but expect any pull request I create will be against the 3.7.2 master branch as would be the UV mapping should that be picked up from hgpovray.         

Related: Yesterday, I think I finally ran down the last part of the bounding control (-MB etc) not working in 3.7 (reported as part of sphere_sweep issue #59) which is tangled some with sphere_sweeps given the long known bounding issues. More testing to do this morning before I create a pull request - probably against the 3.7.1 release branch - for that code change. A working -MB control will make the sphere_sweep testing easier!
